paddle/gserver/layers/AgentLayer.cpp

```diff
@@ -36,14 +36,23 @@ void AgentLayer::forward(PassType passType) {
   Layer::forward(passType);
 
   Argument& realOutput = realLayer_->getOutput();
-  int realHeight = realOutput.getBatchSize();
-  CHECK_LE(numSamples_, realHeight);
+  int realNumSequences = realOutput.getNumSequences();
+  CHECK_LE(numSamples_, realNumSequences);
 
   // get Arguments from real layers
-  if (numSamples_ > 0 && numSamples_ < realHeight) {
-    if (realOutput.ids) {
-      output_.ids =
-          IVector::create(realOutput.ids->getData(), numSamples_, useGpu_);
+  if (numSamples_ > 0 && numSamples_ < realNumSequences) {
+    if (realOutput.hasSeq()) {
```

Comment:
- The original branch `if (realOutput.ids)` is removed, I remember which will be hit in text generation when batch size is changed due to <eos> is generated, and then the codes go to line 57 and `subArgFrom` copies the generated word indices. It seems alright.
- but, I found that the currently the unittest for text generation is over-simplified. It is not what a real text generation process does. It does not include a memory layer that stores the generated word indices and then use the generated word indices to obtain the word embedding. So in the unittest the branch `realOutput.ids` is true seems never be tested.
- The unittest of text generation does not test nested recurrent_layer_group either, so I am not quite sure whether this  PR affect the text generation. I hope to test different network topology after this PR merged.